plugins/module_utils/_hashi_vault_common.py

```diff
@@ -17,22 +17,29 @@
 import os
 
 
-HAS_HVAC = False
-try:
-    import hvac
-    HAS_HVAC = True
-except ImportError:
-    HAS_HVAC = False
-
-
 class HashiVaultValueError(ValueError):
     '''Use in common code to raise an Exception that can be turned into AnsibleError or used to fail_json()'''
 
 
+class HashiVaultHVACError(ImportError):
+    '''Use in common code to signal HVAC is missing.'''
+    def __init__(self, error, msg):
+        super().__init__(error)
+        self.msg = msg
+        self.error = error
+
+
 class HashiVaultHelper():
     def __init__(self):
-        # TODO move hvac checking here?
-        pass
+        try:
+            import hvac
+            self.hvac = hvac
+        except ImportError as e:
+            from ansible.module_utils.basic import missing_required_lib
+            raise HashiVaultHVACError(error=str(e), msg=missing_required_lib('hvac'))
+
+    def get_hvac_exceptions(self):
+        return self.hvac.exceptions
```

Comment:
On one side that was also my first thought. But diving a bit deeper into the code I noticed that only three modules (excluding tests) require the Hvac import.

`vault_pki_generate_certificate.py`, `_auth_method_token.py` and  `_hashi_vault_common.py`. The last is the main player, having the setup for the client and thus also the potential `get_hvac` function. The first only needs Hvac for a constant and the second is a bit tricky with exceptions as well.

My point being that almost all Hvac related things are as good as already handled in the  `_hashi_vault_common.py`. and keeping them there would make it nice and clean. the only thing that's supposedly out of bounds there are the exceptions because they need to be handled on the module level with the json reports.

The trade-off;
having `get_hvac().exceptions` and then the underlying exceptions would have an extra level of mocking potentially. Over, having more possibility to in the future use other sub-categories of hvac.

In the end I have a slight preference over the exceptions function but implementation wise I can do both.

Reply:
@briantist I've quickly looked into the intellisense situation. completion works for the import statement. but from that point (either directly as `hvac.` or `self.hvac.`) it only shows `v1` or `client` and I think that that has to do with the `__init__.py` file. The `hvac.exceptions` are not detected either  way.

![afbeelding](https://github.com/ansible-collections/community.hashi_vault/assets/10096922/efa042b1-d257-4563-bba7-b1f1830b4d76)
Including a python type-hint also does not seem to work even though it seems to detect it as a package. but this brings the extra problem that hvac needs to be imported from outside the function and therefor also needs to have it's 'is installed' tests outside the class. so that resolves to a no-go.

On that note, I did find another way to go over all the exceptions without repeating them in every module.
It might not be the cleanest way but it does tick the box of keeping the hvac module in the `_hashi_vault_common.py`

I've put in two exceptions for now as a trial
![afbeelding](https://github.com/ansible-collections/community.hashi_vault/assets/10096922/106b4be6-c341-49e5-82da-799dbfb5daf3)

This would be the three-way comparison where the first is the original, the second is the `get_hvac().exceptions` or `get_hvac_exceptions()` (both would have virtually identical implementation) and the last would be with a wrapper. 
![afbeelding](https://github.com/ansible-collections/community.hashi_vault/assets/10096922/7d82b034-6dc2-43f7-b089-b925761cd3a2)

Curious about your opinion and final decision implementation wise. Also would be interested in a short overview of what still needs to be done before it is merge-able.